File: leetcode/729. My Calendar I/main.rs
struct MyCalendar {
    events: Vec<(i32, i32)>,
}


/** 
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl MyCalendar {

    fn new() -> Self {
        Self {
            events: vec![],
        }
    }
    
    fn book(&mut self, start: i32, end: i32) -> bool {
        for &evt in &self.events {
            if start < evt.1 && end > evt.0 {
                return false;
            }
        }

        self.events.push((start, end));

        true
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * let obj = MyCalendar::new();
 * let ret_1: bool = obj.book(start, end);
 */

File: leetcode/729. My Calendar I/main.ts
type IEvent = [start: number, end: number]

class MyCalendar {
  private events: Array<IEvent>

  constructor() {
    this.events = []
  }

  book(start: number, end: number): boolean {
    for (const evt of this.events) {
      if (start < evt[1] && end > evt[0]) {
        return false
      }
    }

    this.events.push([start, end])

    return true
  }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * var obj = new MyCalendar()
 * var param_1 = obj.book(start,end)
 */
